watertap/property_models/anaerobic_digestion/adm1_properties.py

```diff
@@ -0,0 +1,346 @@
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Thermophysical property package to be used in conjunction with ADM1 reactions.
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    LiquidPhase,
+    Component,
+    Solute,
+    Solvent,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+import idaes.logger as idaeslog
+
+# Some more information about this module
+__author__ = "Adam Atia"
+# Using Andrew Lee's formulation of ASM1 as a template
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("ADM1ParameterBlock")
+class ADM1ParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = ADM1StateBlock
+
+        # Add Phase objects
+        self.Liq = LiquidPhase()
+
+        # Add Component objects
+        self.H2O = Solvent()
+
+        # All soluble components on kg COD/m^3 basis
+        self.S_su = Solute(doc="Monosaccharides")
+        self.S_aa = Solute(doc="Amino acids")
+        self.S_fa = Solute(doc="Long chain fatty acids")
+        self.S_va = Solute(doc="Total valerate")
+        self.S_bu = Solute(doc="Total butyrate")
+        self.S_pro = Solute(doc="Total propionate")
+        self.S_ac = Solute(doc="Total acetate")
+        self.S_h2 = Solute(doc="Hydrogen gas")
+        self.S_ch4 = Solute(doc="Methane gas")
+        self.S_IC = Solute(doc="Inorganic carbon")
+        self.S_IN = Solute(doc="Inorganic nitrogen")
+        self.S_I = Solute(doc="Soluble inerts")
+
+        self.X_c = Solute(doc="Composites")
+        self.X_ch = Solute(doc="Carbohydrates")
+        self.X_pr = Solute(doc="Proteins")
+        self.X_li = Solute(doc="Lipids")
+        self.X_su = Solute(doc="Sugar degraders")
+        self.X_aa = Solute(doc="Amino acid degraders")
+        self.X_fa = Solute(doc="Long chain fatty acid (LCFA) degraders")
+        self.X_c4 = Solute(doc="Valerate and butyrate degraders")
+        self.X_pro = Solute(doc="Propionate degraders")
+        self.X_ac = Solute(doc="Acetate degraders")
+        self.X_h2 = Solute(doc="Hydrogen degraders")
+        self.X_I = Solute(doc="Particulate inerts")
+
+        # TODO: Additional components not in Table but referred to in text
+        self.S_cat = Solute(doc="Total cation equivalents concentration")
```

Comment:
I haven't read the papers but I wonder if these should really be treated as components, or if they would be better as separate variables on the state blocks.

The answer would lie in how these are calculated - if they are calculated by unit level mass balances (i.e. mass is conserved between states, subject to reaction terms) then they should be components. If however they are calculated based off other concentration (e.g. to maintain a charge balance, which I vaguely recall) then they should not be components (as that would over-specify the problem).